ballerina/soap.bal

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/mime;
+
+# Soap version.
+public type SoapVersion SOAP11 | SOAP12;
+
+public const SOAP11 = "SOAP11";
+public const SOAP12 = "SOAP12";
+
+
+# Object for the basic SOAP client endpoint.
+public client class Client {
+
+    http:Client soapClient;
+    SoapVersion version = SOAP11;
+
+    public function init(string url, http:ClientConfiguration? config = (), SoapVersion? version = ()) returns error? {
```

Comment:
We can make the version as the second argument and make the HTTP client configuration as the rest argument
```bal
    SoapVersion version;

    public function init(string url, SoapVersion version = SOAP11,  *http:ClientConfiguration config) returns error? {
        self.soapClient = check new (url, config);
        self.version = version;
    }
```

This will enable us to
- directly initialize the client without passing the version as `version=SOAP12`
- have the client configurations as named arguments `httpVersion=http:HTTP_1_1`
- show the `SOAP11` is the default version for the client

Reply:
Resolving this as there is a better approach: https://github.com/ballerina-platform/module-ballerina-soap/pull/82#discussion_r1281766958